HGNC_Genomic_Files_Migration/hgnc_data_file_migration.py

```diff
@@ -0,0 +1,243 @@
+
+#---Check the input type (file/directory) and get the list of files (1 or many)--->
+def check_path(source_path):
+	files_list = []
+	exluded_files_list = ['data_bcr_clinical_data_patient.txt','data_bcr_clinical_data_sample.txt','data_clinical_patient.txt','data_clinical_sample.txt','data_clinical_supp_hypoxia.txt','data_gene_matrix.txt','data_microbiome.txt','data_mutational_signature_confidence.txt','data_mutational_signature_contribution.txt','data_timeline_labtest.txt','data_timeline_procedure.txt','data_timeline_specimen.txt','data_timeline_status.txt','data_timeline_surgery.txt','data_timeline_treatment.txt','data_timeline.txt','data_subtypes.txt','data_fusions.txt']
+	
+	if os.path.exists(source_path):
+		if os.path.isdir(source_path):
+			for data_file in os.listdir(source_path):
+				if not data_file.startswith('.') and os.path.isfile(os.path.join(source_path,data_file)) and not data_file.endswith('.seg') and not data_file in exluded_files_list: files_list.append(os.path.join(source_path,data_file))
+		elif os.path.isfile(source_path) and not source_path.endswith('.seg') and not os.path.basename(source_path) in exluded_files_list:
+			files_list.append(source_path)
+		return files_list
+	else:
+		print("ERROR: Invalid path or file '"+source_path+"'")
+		sys.exit(1)
+
+#---If the file has only entrez id and no hugo symbol, check for outdated entrez ids & update--->	
+def update_outdated_entrezids(entrez_index, data_file, outdated_entrez_dict):
+	with open(data_file,'r') as datafile:
+		updated_data = ""
+		log = ""
+		for line in datafile:
+			if line.startswith('#') or line.startswith('Entrez_Gene_Id'):
+				updated_data += line
+			else:
+				line1 = line.strip('\n').split('\t')
+				entrez = line1[entrez_index]
+				if entrez in outdated_entrez_dict:
+					log += entrez+'\t\t---entrez id replaced to---\t\t'+outdated_entrez_dict[entrez]+'\n'
+					line1[entrez_index] = outdated_entrez_dict[entrez]
+					updated_data += '\t'.join(line1)+'\n'
+				else:
+					updated_data += line
+		return updated_data,log
+
+#---If the file has both hugo_symbol and entrez_id columns, update outdated entrez ids first and then update the invalid hugo symbols based on entrez ids--->	
+def update_hugo_symbols(entrez_index, gene_index, data_file, outdated_entrez_dict, outdated_hugo_dict, main_table_entrez_dict, alias_table_entrez_dict):
+	with open(data_file,'r') as datafile:
+		updated_data = ""
+		log = ""
+		for line in datafile:
+			if line.startswith('#') or line.startswith('Entrez_Gene_Id') or line.startswith('Hugo_Symbol'):
+				updated_data += line
+			else:
+				line1 = line.strip('\n').split('\t')
+				entrez = line1[entrez_index]
+				hugo = line1[gene_index]
+				
+				#If entrez id is in outdated list - update the entrez id to new entrez id.
+				if entrez in outdated_entrez_dict:
+					log += entrez+'\t\t---entrez id replaced to---\t\t'+outdated_entrez_dict[entrez]+'\n'
+					line1[entrez_index] = entrez = outdated_entrez_dict[entrez]
```

Comment:
`line1` could be updated to a better variable name.

Reply:
Done